api/controllers/board.js

```diff
@@ -81,20 +81,45 @@ async function getPublicBoards(req, res) {
 
 async function deleteBoard(req, res) {
   const id = req.swagger.params.id.value;
-  Board.findByIdAndRemove(id, function (err, boards) {
+  const board = Board.findOne({ _id: id }, function (err, board) {

```

Comment:
I think this could be done in a simpler way by building the query based on the user's role, and then simply returning a 404 if the board was not found. Most of the existing code would stay the same.

```javascript
const query = req.user.isAdmin ? { _id: id } : { _id: id, email: req.user.email };
Board.findOneAndRemove(query, function (err, board) {
  ...
});
```

(I don't have my personal laptop with me so I haven't tested this.)